algosdk/abi/contract.py

```diff
@@ -10,22 +10,32 @@ class Contract:
 
     Args:
         name (string): name of the contract
-        app_id (int): application id associated with the contract
         methods (list): list of Method objects
+        desc (string, optional): description of the contract
+        networks (dict, optional): information about the contract in a
+            particular network, such as an app-id.
     """
 
-    def __init__(self, name: str, app_id: int, methods: List[Method]) -> None:
+    def __init__(
+        self,
+        name: str,
+        methods: List[Method],
+        desc: str = None,
+        networks: Dict[str, "NetworkInfo"] = None,
+    ) -> None:
         self.name = name
-        self.app_id = int(app_id)
         self.methods = methods
+        self.desc = desc
+        self.networks = networks
```

Comment:
Also, `networks` might be None here, perhaps do `self.networks = networks if networks else {}` instead?